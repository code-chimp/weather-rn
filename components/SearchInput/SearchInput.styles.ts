import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  container: {
    height: 40,
    marginTop: 20,
    marginHorizontal: 40,
    paddingHorizontal: 10,
    borderRadius: 5,
    backgroundColor: '#667',
  },
  textInput: {
    flex: 1,
    color: '#fff',
  },
});

export default styles;
