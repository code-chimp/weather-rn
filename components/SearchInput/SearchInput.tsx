import React, { Component } from 'react';
import { TextInput, View } from 'react-native';
import styles from './SearchInput.styles';

export interface ISearchInputProps {
  placeholder: string;
  onSubmit: (t: string) => void;
}

export interface ISearchInputState {
  text: string;
}

export default class SearchInput extends Component<ISearchInputProps, ISearchInputState> {
  state = {
    text: '',
  };

  handleChangeText = (text: string) => {
    this.setState({ text });
  };

  handleSubmitEditing = () => {
    const { text } = this.state;

    if (!text) {
      return;
    }

    this.props.onSubmit(text);

    this.setState({ text: '' });
  };

  render() {
    return (
      <View style={styles.container}>
        <TextInput
          value={this.state.text}
          onChangeText={this.handleChangeText}
          onSubmitEditing={this.handleSubmitEditing}
          autoCorrect={false}
          placeholder={this.props.placeholder}
          placeholderTextColor="white"
          style={styles.textInput}
          clearButtonMode="always"
        />
      </View>
    );
  }
}
