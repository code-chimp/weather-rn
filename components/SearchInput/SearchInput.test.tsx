import { render, screen } from '@testing-library/react-native';
import SearchInput from './SearchInput';

describe('components/SearchInput', () => {
  it('should match the snapshot', () => {
    render(<SearchInput placeholder="" onSubmit={() => {}} />);

    expect(screen.toJSON).toMatchSnapshot();
  });

  it('should display the supplied placeholder text', async () => {
    const placeholder = 'test placeholder';

    render(<SearchInput placeholder={placeholder} onSubmit={() => {}} />);

    const instances = await screen.findAllByPlaceholderText(placeholder);

    expect(instances).not.toBeNull();
    expect(instances.length).toBeGreaterThan(0);
  });
});
