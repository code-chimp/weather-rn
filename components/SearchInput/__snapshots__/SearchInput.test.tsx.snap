// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/SearchInput should match the snapshot 1`] = `[Function]`;
